api/registrations/serializers.py

```diff
@@ -567,13 +569,55 @@ def _find_orphan_files(self, registering, draft):
         from website.archiver.utils import find_selected_files
         files = find_selected_files(draft.registration_schema, draft.registration_metadata)
         orphan_files = []
-        for _, value in files.items():
+        for key, value in files.items():
             if 'extra' in value:
                 for file_metadata in value['extra']:
-                    if file_metadata['nodeId'] not in registering:
+                    if not self._is_attached_file_valid(file_metadata, registering):
                         orphan_files.append(file_metadata)
+
         return orphan_files
 
+    def _is_attached_file_valid(self, file_metadata, registering):
```

Comment:
make sure this is still being called when reconciled with DraftNode work (new API endpoints there, but they share code with old, so this should still be called).